acvm_backend_barretenberg/src/composer.rs

```diff
@@ -133,31 +159,36 @@ impl StandardComposer {
         unsafe {
             result = Vec::from_raw_parts(proof_addr, proof_size, proof_size);
         }
-        proof::remove_public_inputs(self.constraint_system.public_inputs_size(), &result)
+
+        proof::remove_public_inputs(constraint_system.public_inputs_size(), &result)
     }
 
-    pub fn verify_with_vk(
-        &mut self,
+    fn verify_with_vk(
+        &self,
+        constraint_system: &ConstraintSystem,
         // XXX: Important: This assumes that the proof does not have the public inputs pre-pended to it
         // This is not the case, if you take the proof directly from Barretenberg
         proof: &[u8],
         public_inputs: Assignments,
         verification_key: &[u8],
     ) -> bool {
+        let circuit_size = self.get_circuit_size(constraint_system);
+        let CRS { g2_data, .. } = CRS::new(circuit_size as usize);
```

Comment:
Does this actually use g1_data? Because it might be more efficient to use a `G2::new` instead